### DISCUSSION TITLE
Docker network setup

### DISCUSSION THREAD

**Discussion by davedujour** [automatic-ripping-machine/automatic-ripping-machine#1145] [Q&A]
I followed the instructions on the [wiki for a docker install](https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/docker), but I never saw where the docker container IP address is set. I can't get to the WebU. `ifconfig -a` shows only an IPv6 address for the veth adapter that I assume was setup by docker (i've never setup a virtual ethernet adapter) and most telling of all `docker container list` gives this output:
```
docker container list
CONTAINER ID   IMAGE                                                      COMMAND           CREATED      STATUS                 PORTS                                                 NAMES
30407ec4ff7b   automaticrippingmachine/automatic-ripping-machine:latest   "/sbin/my_init"   3 days ago   Up 9 hours (healthy)   0.0.0.0:8008->8008/tcp, :::8008->8008/tcp, 8080/tcp   arm-ripper
```

So how do I set the IP address in the docker container? It's not in the example config file as the only lines I have changed are `-p` for the port and `-name`.

**Comment by davedujour** [comment]
Here is more network details from `docker container inspect <container-name>`

```         "NetworkMode": "default",
            "PortBindings": {
                "8008/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "8008"
                    }
                ]`
```
This is under config (arm config I think):
```              "Config": {
            "Hostname": "30407ec4ff7b",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "8008/tcp": {},
                "8080/tcp": {}
            },```
and later
```        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "380667e32a619a2ef244e223cbab54c1e41f22e309d9d65e448699a6bf1efc62",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "8008/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "8008"
                    },
                    {
                        "HostIp": "::",
                        "HostPort": "8008"
                    }
                ],
                "8080/tcp": null
            },

**Comment by microtechno9000** [comment]
Docker uses the host machines IP address, so if your server or host running ARM is 192 168.1.1 and the port config is set to 8080. ARM can be accessed at 192.168.1.1:8080

**Comment by davedujour** [comment]
Then something didn't get automagically built correctly in docker when I used the docker installation instructions. I've tried both the default port 8080 and the port I set via the config, in both a browser and doing `telnet 192.168.50.201 8080` and the response is either "connection refused" or "connection timed out." `ifconfig -a` give this:
`ifconfig -a
docker0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 172.17.255.255`
 Which is definitely NOT correct. I don't use 172.x.x.x on my network.
 
 How can I set/reset the docker IP address manually? I don't want to expose this to the public Internet, but I do want to access the ARM WebUI on my internal network so I can make this arm machine headless.

**Reply by microtechno9000** [reply]
what does `docker ps` show?

**Comment by davedujour** [comment]
I think the same info is in `docker container list`, but here is the output of `docker ps`
```
docker ps
CONTAINER ID   IMAGE                                                      COMMAND           CREATED       STATUS                 PORTS                                                 NAMES
83a76ecb8f57   automaticrippingmachine/automatic-ripping-machine:latest   "/sbin/my_init"   2 hours ago   Up 2 hours (healthy)   0.0.0.0:8008->8008/tcp, :::8008->8008/tcp, 8080/tcp   arm-ripper
```

**Reply by microtechno9000** [reply]
they 8008, looks like there might be a typo in your config

**Comment by davedujour** [comment]
I changed the default port of 8080 to 8008. Here's the edited `start_arm_container.sh` file that I used.

```
#!/bin/bash
docker run -d \
    -p "8008:8008" \
    -e ARM_UID="1001" \
    -e ARM_GID="1001" \
    -v "/home/arm:/home/arm" \
    -v "/home/arm/music:/home/arm/music" \
    -v "/home/arm/logs:/home/arm/logs" \
    -v "/home/arm/media:/home/arm/media" \
    -v "/home/arm/config:/etc/arm/config" \
    --device="/dev/sr0:/dev/sr0" \
    --privileged \
    --restart "always" \
    --name "arm-ripper" \
    --cpuset-cpus='1,2,3' \
    automaticrippingmachine/automatic-ripping-machine:latest
````

To clarify what I said about an hour before this, I've tried hitting both the default port 8080 and my custom port 8008. Neither works. Shouldn't there be an IP address in place either the first or second part of this? `0.0.0.0:8008->8008/tcp, :::8008->8008/tcp,` Like `192.168.1.10:8008->8008/tcp` or similar?

**Comment by davedujour** [comment]
I think I figured it out. When I change `-p "8008:8008" \ ` to use a custom port, I changed both the host side & docker side ports. I figured this would be fine, shouldn't arm just listen on whatever port is specified? I went back to the default ports and it works. For a custom port (say 7000) to access arm, should that be `-p "7000:8080"?

**Reply by microtechno9000** [reply]
Yes, that will do it
For reference the port setup can be found here

https://docs.docker.com/guides/docker-concepts/running-containers/publishing-ports/

**Comment by davedujour** [comment]
Thanks for the help!
